types/react-router/index.d.ts

```diff
@@ -164,3 +165,36 @@ export function useRouteMatch<Params extends { [K in keyof Params]?: string } =
 export function useRouteMatch<Params extends { [K in keyof Params]?: string } = {}>(
     path: string | string[] | RouteProps,
 ): match<Params> | null;
+
+export interface RouteRenderConfig {
+    path: string;
+    element?: React.ReactNode;
+    children: RouteConfig[];
+}
+
+export interface RouteRenderWithChildrenConfig {
+    path: string;
+    element: React.ReactNode;
+}
+
+export interface RouteRedirectConfig {
+    path: string;
+    redirectTo: string;
+}
+
+export type RouteConfig = RouteRenderConfig | RouteRenderWithChildrenConfig | RouteRedirectConfig;
+
+/**
+ * Returns the element of the route that matched the current location, prepared
+ * with the correct context to render the remainder of the route tree. Route
+ * elements in the tree must render an <Outlet> to render their child route's
+ * element.
+ *
+ * Route objects may take one of 2 forms:
+ *
+ * - `{ path, element?, children }`
+ * - `{ path, redirectTo }`
+ * 
+ * If no `element` is provided, `<Outlet />` will be used implicitly.
+ */
```

Comment:
I appreciate the explanation, but I would say this is not a place for comments, so I would remove it.

Reply:
I wish these docs were included on all functions as they enable really nicely formatted hover docs in editors that support LSPs.

![image](https://user-images.githubusercontent.com/18172185/80162442-389bc300-8588-11ea-820c-897c8c391c21.png)

If we add them it should be for all, not just this function, so I'm cool with removing them for now.

FWIW it looks like the `react-router` devs agree - https://github.com/ReactTraining/react-router/blob/723fb3b2b95f840981d7da4af3bd625e01997399/packages/react-router/index.js#L468